Sources/Overlays/_Testing_CoreGraphics/Attachments/_AttachableImageContainer.swift

```diff
@@ -0,0 +1,163 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if SWT_TARGET_OS_APPLE && canImport(CoreGraphics)
+@_spi(Experimental) public import Testing
+private import CoreGraphics
+
+private import ImageIO
+import UniformTypeIdentifiers
+
+/// ## Why can't images directly conform to Attachable?
+///
+/// Three reasons:
+///
+/// 1. Several image classes are not marked `Sendable`, which means that as far
+///    as Swift is concerned, they cannot be safely passed to Swift Testing's
+///    event handler (primarily because `Event` is `Sendable`.) So we would have
+///    to eagerly serialize them, which is unnecessarily expensive if we know
+///    they're actually concurrency-safe.
+/// 2. We would have no place to store metadata such as the encoding quality
+/// 	 (although in the future we may introduce a "metadata" associated type to
+///    `Attachable` that could store that info.)
+/// 3. `Attachable` has a requirement with `Self` in non-parameter, non-return
+///    position. As far as Swift is concerned, a non-final class cannot satisfy
+///    such a requirement, and all image types we care about are non-final
+///    classes. Thus, the compiler will steadfastly refuse to allow non-final
+///    classes to conform to the `Attachable` protocol. We could get around this
+///    by changing the signature of `withUnsafeBufferPointer()` so that the
+///    generic parameter to `Attachment` is not `Self`, but that would defeat
+///    much of the purpose of making `Attachment` generic in the first place.
+///    (And no, the language does not let us write `where T: Self` anywhere
+///    useful.)
+
+/// A wrapper type for image types such as `CGImage` and `NSImage` that can be
+/// attached indirectly.
+///
+/// You do not need to use this type directly. Instead, initialize an instance
+/// of ``Attachment`` using an instance of an image type that conforms to
+/// ``AttachableAsCGImage``. The following system-provided image types conform
+/// to the ``AttachableAsCGImage`` protocol and can be attached to a test:
+///
+/// - [`CGImage`](https://developer.apple.com/documentation/coregraphics/cgimage)
+@_spi(Experimental)
+public struct _AttachableImageContainer<ImageClass>: Sendable where ImageClass: AttachableAsCGImage {
```

Comment:
Any reason "Class" needs to be in the generic type parameter `ImageClass`, or could it just be `Image`?

Reply:
I think at one point I had it constrained to `AnyObject`.